### PR TITLE
Allow deleting merged branches

### PR DESCRIPTION
**Pull request by mmcclimon**
I mean honestly, GitLab. I need to test this a bit more.

### CODE REVIEW

**Comment by mmcclimon** [comment]
@rjbs This is two commits, the first just code shuffling and the second actually interesting.

**Review by rjbs — APPROVED** [review]
Weird, I thought your "wait until noticed merged" was already in!  But looks good.

**Pull request merged**
